projects/igniteui-angular/src/lib/grids/columns/column.component.ts

```diff
@@ -1422,7 +1422,8 @@ export class IgxColumnComponent implements AfterContentInit, OnDestroy, ColumnTy
         return this.grid.dataView
             .map((rec, index) => {
                 if (!this.grid.isGroupByRecord(rec) && !this.grid.isSummaryRow(rec)) {
-                    this.grid.pagingMode === 1 && this.grid.page !== 0 ? index = index + this.grid.perPage * this.grid.page : index = this.grid.dataRowList.first.index + index;
+                    this.grid.pagingMode === GridPagingMode.Remote && this.grid.page !== 0 ?
```

Comment:
Just use the actual value, the constant is there mostly for backwards compat
```suggestion
                    this.grid.pagingMode === 'remote' && this.grid.page !== 0 ?
```
Same for the other places ofc